pmm-app/src/pmm-settings/Parts/Settings/Settings.tsx

```diff
@@ -176,20 +195,20 @@ const SettingsPart = props => {
                       />
                     }
                     element={
-                      <div style={{ display: 'flex', justifyContent: 'flex-start', width: '100%' }}>
-                        <InputField name={'data_retention_count'} form={form} wrapperStyle={{ width: '60%' }} />
-                        <SelectField
-                          form={form}
-                          name={'data_retention_units'}
-                          options={dataRetentionOptions}
-                          defaultValue={'weeks'}
-                          style={{ width: '40%' }}
-                        />
-                      </div>
+                      <InputField
+                        name={'data_retention_count'}
+                        validate={Validators.compose(
+                          Validators.range(1, 3650),
+                          Validators.required
```

Comment:
`required` should come first, anything else should almost always follow, not vice versa.
It just doesn't make much sense to check the value if there in no value.
See an example [here](https://github.com/final-form/react-final-form/blob/master/examples/field-level-validation/index.js#L55).